### PR TITLE
[G2M]Port/react labs

### PR DESCRIPTION
**Pull request by tlowande**
references: 
- EQWorks/flaat-dash#163
- [Splitting one Git Repo into Multiple Repos](https://www.youtube.com/watch?v=P5r_jONyuHo)
- [Move files from one repository to another, preserving git history](https://medium.com/@ayushya/move-directory-from-one-repository-to-another-preserving-git-history-d210fa049d4b)
- [Tips for subtree splitting and moving](https://medium.com/@slhenty/git-tips-for-subtree-splitting-and-moving-560f3743e19c)
- [Splitting up a Git repo](http://jimmy.schementi.com/splitting-up-a-git-repo)

log of git operations done on extraction:
``` bash
 #  on react-labs: 
> git fetch --prune
> git remote rm origin
> git checkout -b extraction
# [next step is optional to put together list of files to keep]
# list each portion
> ls -la src
# then copy the list result to vscode
# hold `option` key and click  in the desired position to add cursor to all lines. Delete everything that is not the actual file name. Add path if needed (like src/ or stories/ ...) 
# select all and `CTRL + j` to make it one line 
# [end of optional part]
# export a string of files to keep as a var
> export PATHS_TO_KEEP=".depcheckrc .eslintrc.js .nvmrc .github babel.config.js netlify-dp-ignore.sh netlify.toml.storybook src/alert.js src/button.js src/checkbox.js src/chip.js src/list src/loader src/radio.js src/switch.js src/tab-panels.js src/text-field.js src/theme src/typography.js src/with-ref.js stories/alert.stories.js stories/button.stories.js stories/checkbox.stories.js stories/chip.stories.js stories/list.stories.js stories/loader.stories.js stories/radio.stories.js stories/switch.stories.js stories/tab-panels.stories.js stories/text-field.stories.js stories/typography.stories.js"
# now filter the branch
> git filter-branch -f --index-filter "git rm --ignore-unmatch --cached -qr -- . && git reset -q \$GIT_COMMIT -- $PATHS_TO_KEEP" --prune-empty -- HEAD
# there will be a warning so the process takes a few min to initiate. Make sure it is sucessful
```

``` bash
# then on lumen-ui side
(lumen-ui/master)% fetch --prune && git pull
(lumen-ui /master)% git checkout -b port/react-labs
(lumen-ui /port/react-labs)% git pull ../react-labs extraction --allow-unrelated-histories
# clean up things, squash unwanted commits, add setup files, add story ...
# git add . 
# git commit --amend 
# and push :)
```
Screen Shot 2021-01-20 at 12 34 13 PM
Screen Shot 2021-01-20 at 12 34 33 PM

### CODE REVIEW

**Comment by tlowande** [comment]
@woozyking not sure if I have all the needed files, but the storybook seems to be working :) can you check dependencies?
also, I kept merge commits. Should I get rid of them?

Also, I brought theme here, but eventually, this will be extracted out as well right?

**Comment by woozyking** [comment]
> also, I kept merge commits. Should I get rid of them?

no harm in keeping them 👍 

> Also, I brought theme here, but eventually, this will be extracted out as well right?

yes, _ideally_ we should be able to offer `lumen-ui` as-is without themes to at least _function_ (may not be optimal in look/feels). but this is going to be a quest for us to find a solution collectively. before then we can stick with the current state

**Review by woozyking — APPROVED** [review]
looks good. may have small issues with github actions but should be addressable quickly when we see them (so long locally the fundamental steps work as expected)

minor things:
* `master.yml` should probably be renamed as `main.yml` to reflect the branch it acts on
* netlify configs as discussed can probably be dropped

File: package.json [hunk 1]
```diff
@@ -0,0 +1,75 @@
+{
+  "name": "lumen-ui",
```
Comment (woozyking) [inline package.json]: forgot that this should be named `@eqworks/lumen-ui` to be a scoped package

**Review by woozyking — APPROVED** [review]

**Pull request merged**